File: android/app/src/main/kotlin/nl/rijksoverheid/edi/wallet/hwkeystore/bridge/SigningKey.kt
package nl.rijksoverheid.edi.wallet.hwkeystore.bridge

import uniffi.hw_keystore.SigningKeyBridge

class SigningKey : SigningKeyBridge {
    override fun publicKey(): List<UByte> {
        val temp = "publicKey"
        return temp.toByteArray().toList() as List<UByte>
    }

    override fun sign(payload: List<UByte>): List<UByte> {
        val temp = "sign"
        return temp.toByteArray().toList() as List<UByte>
    }
}

File: wallet_core/wallet/platform_support/ios/PlatformSupport/HWKeyStore/Bridge/SigningKey.swift
//
//  SigningKey.swift
//  HWKeyStore
//
//  Created by Wallet Developer on 24/02/2023.
//

import Foundation

final class SigningKey {
    private static let identifierPrefix = "ecdsa"

    private func secureEnclaveKey(for identifier: String) throws -> SecureEnclaveKey {
        return try SecureEnclaveKey(identifier: "\(Self.identifierPrefix)_\(identifier)")
    }
}

extension SigningKey: SigningKeyBridge {
    func publicKey(identifier: String) throws -> [UInt8] {
        do {
            return try Array(self.secureEnclaveKey(for: identifier).encodePublicKey())
        } catch let error as SecureEnclaveKeyError {
            throw KeyStoreError.from(error)
        }
    }

    func sign(identifier: String, payload: [UInt8]) throws -> [UInt8] {
        do {
            return try Array(self.secureEnclaveKey(for: identifier).sign(payload: Data(payload)))
        } catch let error as SecureEnclaveKeyError {
            throw KeyStoreError.from(error)
        }
    }

    func delete(identifier: String) throws {
        do {
            return try self.secureEnclaveKey(for: identifier).delete()
        } catch let error as SecureEnclaveKeyError {
            throw KeyStoreError.from(error)
        }
    }
}
